--- a/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_vi.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_mqd_manager_vi.c
@@ -175,7 +175,8 @@

 	m->cp_hqd_pq_control = 5 << CP_HQD_PQ_CONTROL__RPTR_BLOCK_SIZE__SHIFT |
 			atc_bit << CP_HQD_PQ_CONTROL__PQ_ATC__SHIFT |
-			mtype << CP_HQD_PQ_CONTROL__MTYPE__SHIFT;
+			mtype << CP_HQD_PQ_CONTROL__MTYPE__SHIFT |
+			CP_HQD_PQ_CONTROL__UNORD_DISPATCH_MASK;
 	m->cp_hqd_pq_control |=	order_base_2(q->queue_size / 4) - 1;
 	pr_debug("cp_hqd_pq_control 0x%x\n", m->cp_hqd_pq_control);
